lib/new_relic/agent/tracer.rb

```diff
@@ -410,9 +410,17 @@ def clear_state
 
         alias_method :tl_clear, :clear_state
 
+        def current_segment_key
+          ::Fiber.current.object_id
```

Comment:
Using the fiber id instead of the thread id (which is what we did before), still totally works with threads, because every new thread is a new fiber as well, but it also allows us to accurately track multiple fibers that exist inside of a single thread.